packages/teams-js/src/public/dialog.ts

```diff
@@ -33,6 +33,23 @@ export namespace dialog {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   export type PostMessageChannel = (message: any) => void;
   export type DialogSubmitHandler = (result: ISdkResponse) => void;
+  const storedMessages: string[] = [];
+
+  /**
+   * @hidden
+   * Hide from docs because this function is only used during initialization
+   * ------------------
+   * Adds register handlers for messageForChild upon initialization and only in the tasks FrameContext. {@link FrameContexts.task}
+   * Function is called during app intitialization
+   * @internal
+   */
+  export function initialize(): void {
+    ensureInitialized(FrameContexts.task);
```

Comment:
This will throw if framecontext is not `task`. That's not a good thing in this function, because `initialize` is called for every app, not just dialogs.

Looking at the other capabilities with `initialize` none of them uses `ensureInitialized`. So I think you shouldn't use it here either.

Reply:
Well, sorry, I should have been more specific. I think we should keep a check and only register if frameContext is `task`, but it can't be done via `ensureInitialized` because that throws and that's not what we want.

Reply:
Oh don't worry, I picked that up on that. I'm just testing different checks right now. I tried doing `GlobalVars.frameContext === FrameContext.tasks`, unfortunately since GlobalVars isn't set at the time this leads to the handler not being set. My next approach is to do something similar to `menus.initialize` and have a helper function.

Reply:
I've added in the helper function